api/hooks.ts

```diff
@@ -1,25 +1,237 @@
 import { APIConfig, FastifyTypebox } from '.'
-import Store from '../store'
-
-export const hookRoutes = (cfg: APIConfig, store: Store) => async (server: FastifyTypebox): Promise<void> => {
-  // Register hooks for new inbox items that get added to the moderation queue
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-  server.delete('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-  server.put('/:domain/hooks/onmoderationqueued', async (request, reply) => { })
-
-  // Register hooks for new inbox items that get approved
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onnew', async (request, reply) => { })
-  server.delete('/:domain/hooks/onnew', async (request, reply) => { })
-  server.put('/:domain/hooks/onnew', async (request, reply) => { })
-
-  // Register hooks for new inbox items that get rejected
-  // Hooks are an array of {url, method, headers}
-  // The body will contain the inbox item
-  server.get('/:domain/hooks/onrejected', async (request, reply) => { })
-  server.delete('/:domain/hooks/onrejected', async (request, reply) => { })
-  server.put('/:domain/hooks/onrejected', async (request, reply) => { })
+import { Type } from '@sinclair/typebox'
+import { HookStore } from '../store/HookStore'
+
+export const hookRoutes = (cfg: APIConfig, store: HookStore) => async (server: FastifyTypebox): Promise<void> => {
+  // For ModerationQueued
+  server.put('/:actor/hooks/moderationqueued', {
+    schema: {
+      params: Type.Object({
+        actor: Type.String()
+      }),
+      body: Type.Object({
+        url: Type.String({
+          format: 'uri'
+        }),
+        method: Type.Union([Type.Literal('GET'), Type.Literal('POST'), Type.Literal('PUT'), Type.Literal('DELETE')]),
```

Comment:
Can you take this type and put it into it's own const at the top called `WebHookSchema`? Maybe place the type definition into hookstore or hooksystem.